File: 04.Simulation模拟/leetcode_2996_missingInteger.py
from typing import List
from collections import Counter

class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        cnt = Counter(nums)
        nums.append(-1)
        mx = nums[0]
        i = 1
        while nums[i-1]+1 == nums[i]:
            mx += nums[i]
            i += 1
        while mx in cnt:
            mx += 1
        return mx

if __name__ == "__main__":
    s = Solution()
    nums = [1,2,3,2,5]
    print(s.missingInteger(nums))

File: 04.Simulation模拟/leetcode_2996_missingInteger.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int missingInteger(vector<int>& nums) {
        unordered_map<int, int> cnt;
        for (int i : nums) cnt[i]++;
        nums.push_back(-1);
        int i = 1, mx = nums[0];
        while (nums[i-1]+1 == nums[i]){
            mx += nums[i];
            i++;
        }

        while (cnt.count(mx)){
            mx++;
        }
        return mx;
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,2,3,2,5};
    cout << s.missingInteger(nums) << endl;
}
